WebDriverAgentLib/Categories/XCUIElement+FBFind.m

```diff
@@ -13,21 +13,32 @@
 #import "FBElementTypeTransformer.h"
 #import "XCElementSnapshot.h"
 #import "XCElementSnapshot+FBHelpers.h"
+#import "XCUIElement+FBUtilities.h"
 #import "XCUIElement+FBWebDriverAttributes.h"
 
 @implementation XCUIElement (FBFind)
 
 
 #pragma mark - Search by ClassName
 
-- (NSArray<XCUIElement *> *)fb_descendantsMatchingClassName:(NSString *)className
+- (NSArray<XCUIElement *> *)fb_descendantsMatchingClassName:(NSString *)className shouldReturnAfterFirstMatch:(BOOL)shouldReturnAfterFirstMatch
```

Comment:
actually it does, if you check the actual implementation. The actual query object is "lazy" (only calculated upon calling):
![scr](https://cloud.githubusercontent.com/assets/7767781/20094866/69e21464-a5a3-11e6-9f2b-255e0d3ea4cd.png)

Which means that _elementBoundByIndex_ call should not query all the objects, but rather just one of them. I've measured the actual performance there and this call is really faster, but not so much. So I reported another bug to Apple regarding the actual performance of 
_allElementsBoundByIndex_
and
_elementBoundByIndex_
calls, since I expect the second one to be much faster, because we don't have to enumerate all the descendant elements, but rather only one in our case.